engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -371,6 +360,50 @@ class Print : public PhysicalPlan {
 	std::ostream * ofs = nullptr;
 	std::mutex print_lock;
 };
+ 
+
+class OutputKernel : public kernel {
+public:
+	OutputKernel() : kernel("OutputKernel") {  }
+	virtual kstatus run() {
+		output = std::move(this->input_.get_cache()->pullFromCache());
```

Comment:
Why does this run only need to operate once? Is that becuase it is connected to some special cache that will concat all the subsets first?

Reply:
Yes the OutputKernel uses a ConcatenatingCacheMachine.